### PR TITLE
Mods for subscribe function

### PR DESCRIPTION
**Pull request by evgeny-s**
* Updated subscribe function
* Created AppSubscription
* Removed balances
* Moved balances under the AppSubscription
* Test coverage

### CODE REVIEW

File: cere02/lib.rs [hunk 1]
```diff
@@ -734,6 +790,19 @@ mod ddc {
             let payer = AccountId::from([0x1; 32]);
             assert_eq!(contract.balance_of(payer), 0);
             assert_eq!(contract.subscribe(3), Ok(()));
+
+            let mut subscription = contract.subscriptions.get(&payer).unwrap();
+
+            assert_eq!(subscription.end_date_ms, 31 * MS_PER_DAY);
+            assert_eq!(subscription.balance, 500);
+
+            contract.subscribe(3);
```
Comment (naure) [inline cere02/lib.rs:799]: `.unwrap()`

File: cere02/lib.rs [hunk 2]
```diff
@@ -402,13 +428,21 @@ mod ddc {
 
         /// Return balance of an account
         fn balance_of_or_zero(&self, owner: &AccountId) -> Balance {
-            *self.balances.get(owner).unwrap_or(&0)
+            let subscription_opt = self.subscriptions.get(&owner);
```
Comment (naure) [inline cere02/lib.rs:431]: (minor)
This is identical to balance_of, remove it or make one call the other.

File: cere02/lib.rs (hunk 1 above)
Comment (evgeny-s) [reply cere02/lib.rs:799]: What is the difference between this and unwrap?
Tests are working as expected without this. Could you please explain?

**Comment by evgeny-s** [comment]
@naure 
Fixed comments

**Pull request merged**